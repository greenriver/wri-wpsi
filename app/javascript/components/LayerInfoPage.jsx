import React from 'react'
import { withRouter } from 'react-router-dom'
import Modal from 'components/Modal'

const LayerInfoPage = ({ match, history }) => {
  const { params } = match

  const layerId = params.layerId
  const layer = window.layers.find(layer => layer.id === layerId)

  return (
    <Modal>
      <h1>{layer.name}</h1>

      <p>
        {layer.long_description || 'A long description of this dataset is not yet available.'}
      </p>

      <button onClick={() => history.goBack()}>Close</button>
    </Modal>
  )
}

import PropTypes from 'prop-types'
LayerInfoPage.propTypes = {
  match: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
}

export default withRouter(LayerInfoPage)
