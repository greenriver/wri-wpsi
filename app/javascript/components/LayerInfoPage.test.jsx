import React from 'react'
import { MemoryRouter, Route } from 'react-router-dom'
import LayerInfoPage from 'components/LayerInfoPage'
import renderer from 'react-test-renderer'

const layers = [
  {
    id: '1',
    name: 'Test name',
    long_description: 'Test description',
  },
  {
    id: '2',
    name: 'Test name',
  }
]

const checkRenderWithLayer = (layerId) => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[`/map/learn_more/${layerId}`]}>
      <Route
        path='/map/learn_more/:layerId'
        render={() => <LayerInfoPage layers={layers} />}
      />
    </MemoryRouter>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

it('renders correctly', () => {
  checkRenderWithLayer('1')
})

it('renders correctly without a description too', () => {
  checkRenderWithLayer('2')
})
