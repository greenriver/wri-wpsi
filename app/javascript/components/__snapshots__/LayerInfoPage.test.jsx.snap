// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="modal-background"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.4)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "left": "0",
      "overflow": "auto",
      "position": "fixed",
      "top": "0",
      "width": "100%",
      "zIndex": "10001",
    }
  }
>
  <div
    id="modal"
    style={
      Object {
        "backgroundColor": "#fefefe",
        "border": "1px solid #888",
        "borderRadius": 4,
        "boxShadow": "0 2px 10px 1px rgba(0,0,0, .1)",
        "maxHeight": "80vh",
        "maxWidth": "1000px",
        "overflow": "hidden",
        "padding": "30px 20px 20px 20px",
        "width": "80vw",
      }
    }
  >
    <header
      className="LayerInfoPage-header-0-1-1"
    >
      <h1>
        Test name
      </h1>
      <button
        aria-label="Close"
        className="LayerInfoPage-closeButton-0-1-3"
        onClick={[Function]}
      >
        <svg
          className="wri_api__c-icon wri_api__-small"
          style={Object {}}
        >
          <use
            xlinkHref="#icon-cross"
          />
        </svg>
      </button>
    </header>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "Test description",
        }
      }
    />
    <button
      className="LayerInfoPage-button-0-1-2"
      onClick={[Function]}
    >
      Close
    </button>
  </div>
</div>
`;

exports[`renders correctly without a description too 1`] = `
<div
  id="modal-background"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.4)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "left": "0",
      "overflow": "auto",
      "position": "fixed",
      "top": "0",
      "width": "100%",
      "zIndex": "10001",
    }
  }
>
  <div
    id="modal"
    style={
      Object {
        "backgroundColor": "#fefefe",
        "border": "1px solid #888",
        "borderRadius": 4,
        "boxShadow": "0 2px 10px 1px rgba(0,0,0, .1)",
        "maxHeight": "80vh",
        "maxWidth": "1000px",
        "overflow": "hidden",
        "padding": "30px 20px 20px 20px",
        "width": "80vw",
      }
    }
  >
    <header
      className="LayerInfoPage-header-0-1-1"
    >
      <h1>
        Test name
      </h1>
      <button
        aria-label="Close"
        className="LayerInfoPage-closeButton-0-1-3"
        onClick={[Function]}
      >
        <svg
          className="wri_api__c-icon wri_api__-small"
          style={Object {}}
        >
          <use
            xlinkHref="#icon-cross"
          />
        </svg>
      </button>
    </header>
    <p
      dangerouslySetInnerHTML={
        Object {
          "__html": "A long description of this dataset is not yet available.",
        }
      }
    />
    <button
      className="LayerInfoPage-button-0-1-2"
      onClick={[Function]}
    >
      Close
    </button>
  </div>
</div>
`;
